chia/consensus/blockchain.py

```diff
@@ -260,6 +232,8 @@ async def receive_block(
             npc_result,
             fork_point_with_peak,
             self.get_block_generator,
+            # If we did not already validate the signature, validate it now
+            validate_signature=pre_validation_result.validated_signature,
```

Comment:
didn't you mean to negate `pre_validation_result.validated_signature`?

Reply:
"validate_signature" sounds like asking the function (`validate_block_body()`) to perform the validation, and `validated_signature` sounds like it's saying whether the signature has been validated already.

Reply:
wow yes we need to negate it. What a huge mistake!

Reply:
it would be great to have a test that would catch this

Reply:
Roger that